source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h

```diff
@@ -15,22 +19,50 @@ namespace LocalRateLimit {
 
 class LocalRateLimiterImpl {
 public:
-  LocalRateLimiterImpl(const std::chrono::milliseconds fill_interval, const uint32_t max_tokens,
-                       const uint32_t tokens_per_fill, Event::Dispatcher& dispatcher);
+  LocalRateLimiterImpl(
+      const std::chrono::milliseconds fill_interval, const uint32_t max_tokens,
+      const uint32_t tokens_per_fill, Event::Dispatcher& dispatcher,
+      const Envoy::Protobuf::RepeatedPtrField<
+          envoy::extensions::common::ratelimit::v3::LocalRateLimitDescriptor>& descriptors,
+      bool per_route);
   ~LocalRateLimiterImpl();
 
   bool requestAllowed() const;
 
+  bool areDescriptorsConfigured() const;
```

Comment:
I don't think this function is needed. It would be simpler to just have the requestAllowed() function return false or an enum saying no descriptors if that is actually required.

Reply:
Added this because we wanted to do check for descriptors only if they were configured and as we moved parsing of descriptors to local rate limit.. this function moved here too

Reply:
I don't think this flow makes sense and you can simplify. The filter should be able to pass all (optional) data into the limiter and have it to what it needs to do. Please simplify.

Reply:
let me look at this again...

Reply:
Fixed